cmd/internal/backup/providers/gcp/gcp.go

```diff
@@ -101,18 +101,39 @@ func New(ctx context.Context, log *slog.Logger, config *BackupProviderConfigGCP)
 // EnsureBackupBucket ensures a backup bucket at the backup provider
 func (b *BackupProviderGCP) EnsureBackupBucket(ctx context.Context) error {
 	bucket := b.c.Bucket(b.config.BucketName)
-	lifecycle := storage.Lifecycle{
-		Rules: []storage.LifecycleRule{
-			{
-				Condition: storage.LifecycleCondition{
-					NumNewerVersions: b.config.ObjectsToKeep,
-				},
-				Action: storage.LifecycleAction{
-					Type: "Delete",
-				},
-			},
+
+	rule := storage.LifecycleRule{
+		Condition: storage.LifecycleCondition{
+			NumNewerVersions: b.config.ObjectsToKeep,
+			MatchesPrefix:    []string{b.config.ObjectPrefix},
+		},
+		Action: storage.LifecycleAction{
+			Type: "Delete",
 		},
 	}
+
+	// get existing lifecycle configuration
+	bucketAttrs, err := bucket.Attrs(ctx)
+	if bucketAttrs != nil {
+		if err != nil {
+			return err
+		}
+
+		for _, rule := range bucketAttrs.Lifecycle.Rules {
+			if rule.Condition.MatchesPrefix != nil {
+				for _, prefix := range rule.Condition.MatchesPrefix {
+					if prefix != b.config.ObjectPrefix {
+						rule.Condition.MatchesPrefix = append(rule.Condition.MatchesPrefix, prefix)
+					}
+				}
+			}
+		}
+	}
+
+	lifecycle := storage.Lifecycle{
+		Rules: []storage.LifecycleRule{rule},
```

Comment:
This just sets a single rule and also for some reason this does only include a single condition (the one of the last started sidecar). I think you should create multiple rules as otherwise you cannot specify different `NumNewVersions`. So just iterate through the rules and take all of them, if your prefix is contained, update it otherwise add it.